improver/nowcasting/optical_flow.py

```diff
@@ -844,18 +887,18 @@ def process(self, cube1, cube2, boxsize=30):
             # convert displacements to velocities in metres per second
             for vel in [ucomp, vcomp]:
                 vel *= np.float32(1000.0 * grid_length_km)
-                vel /= cube_time_diff.total_seconds()
+                vel /= time_diff_seconds
 
         # create velocity output cubes based on metadata from later input cube
-        x_coord = cube2.coord(axis="x")
-        y_coord = cube2.coord(axis="y")
-        t_coord = cube2.coord("time")
         ucube = iris.cube.Cube(
```

Comment:
I'm surprised there aren't `forecast_reference_time` and `forecast_period` coords being added to these output cubes. Without them, we won't be able to tell which was the original (observation - observation) and which was the perturbation (forecast - observation).

Reply:
The times are different.  The initial advection velocity components were calculated on the previous cycle, so they have the time of the previous cycle.  The forecast period is zero and the "forecast reference time" (if one were relevant - since this is an observation I don't think forecast reference time is applicable at all) is equal to the value of the validity time coordinate.